src/Propel/Runtime/Collection/Collection.php

```diff
@@ -201,7 +201,7 @@ public function setData(array $data): void
     }
 
     /**
-     * @return \Propel\Runtime\Collection\CollectionIterator
+     * @return \Propel\Runtime\Collection\CollectionIterator|\Propel\Runtime\Collection\IteratorInterface
```

Comment:
With
> class CollectionIterator extends ArrayIterator implements IteratorInterfac
is this still needed here then?

Reply:
OnDemandIterator is not a subtype of Collection iterator. So I can leave only interface here, but those iterator classes are quite different under the hood, because Collection one implements ArrayIterator and OnDemand one - not